cadc-web-util/src/main/java/ca/nrc/cadc/accesscontrol/AccessControlClient.java

```diff
@@ -95,22 +104,106 @@ public void resetPassword(final char[] newPassword, final char[] token) {
+            Subject.doAs(subject, new PrivilegedExceptionAction<Void>() {
+
+                @Override
+                public Void run() throws Exception {
+                    final HttpPost thePost = postNoRedirect(lookupPasswordResetURL(), payload, headers);
+                    int statusCode = thePost.getResponseCode();
+                    StringBuilder logStr = new StringBuilder();
+                    logStr.append("Unable to reset password");
+                    StringBuilder throwStr = new StringBuilder();
+                    throwStr.append("Unable to reset password");
+
+                    // Gets added after specific message
+                    StringBuilder reasonPart = new StringBuilder();
+                    if (statusCode != 200) {
+                        reasonPart.append(" - ");
+                        reasonPart.append(thePost.getResponseCode());
+                        reasonPart.append(": ");
+                        reasonPart.append(thePost.getThrowable().toString());
+                    }
+                    String msg = "";
+
+                    switch (statusCode) {
+                        case 200: {
+                            break;
+                        }
+                        case 400: {
+                            throwStr.append(": ");
+                            throwStr.append(thePost.getThrowable().getMessage());
+                            logStr.append(msg);
+                            logStr.append(reasonPart.toString());
+                            log.error(logStr.toString());
+                            throw new IllegalArgumentException(throwStr.toString());
+                        }
+                        case 403:
+                        case 401: {
+                            msg = ": Login denied";
+                            throwStr.append(msg);
+                            logStr.append(msg);
+                            logStr.append(reasonPart.toString());
+                            log.error(logStr.toString());
+                            throw new AccessControlException(throwStr.toString());
+                        }
+                        case 404: {
+                            msg = ": Service unavailable";
+                            throwStr.append(msg);
+                            logStr.append(msg);
+                            logStr.append(reasonPart.toString());
+                            log.error(logStr.toString());
+                            throw new IllegalArgumentException(throwStr.toString());
+                        }
+                        case -1: {
+                            throwStr.append(": Bad request");
+                            logStr.append(": Call not completed");
+                            logStr.append(reasonPart.toString());
+                            log.error(logStr.toString());
+                            throw new IllegalArgumentException(throwStr.toString());
```

Comment:
This should be IllegalStateException since it results from a bug somewhere in our code.